### PR TITLE
Misc documentation fixes

### PR DESCRIPTION
**Pull request by teobiton**

### CODE REVIEW

File: src/cocotb/clock.py [hunk 1]
```diff
@@ -119,15 +120,17 @@ def __init__(
         self.coro = None
         self.mcoro = None
 
-    async def start(self, cycles=None, start_high=True):
+    async def start(
+        self, cycles: Optional[int] = None, start_high: Optional[bool] = True
```
Comment (ktbarrett) [inline src/cocotb/clock.py:124]:
```suggestion
        self, cycles: Optional[int] = None, start_high: bool = True
```
Here optional just means that it's not required, i.e. you don't have to pass an argument, not that it's `typing.Optional` AKA `X | None`.

Comment (teobiton) [reply src/cocotb/clock.py]: Ah, I misunderstood the original comment. Fixed it!

**Review by ktbarrett — APPROVED** [review]

**Comment by teobiton** [comment]
Oops, I referenced the wrong issue number in the commit, I'll force push again.

**Comment by ktbarrett** [comment]
I don't think this is a very contentious change, so I won't wait up.

**Pull request merged**